dask/dataframe/core.py

```diff
@@ -4461,9 +4461,12 @@ def hash_shard(df, nparts, split_out_setup=None, split_out_setup_kwargs=None):
         h = split_out_setup(df, **(split_out_setup_kwargs or {}))
     else:
         h = df
-    h = hash_pandas_object(h, index=False)
+    import sys
+
+    package = sys.modules[df.__class__.__module__.split(".")[0]]
+    h = package.util.hash_pandas_object(h, index=False)
     if is_series_like(h):
-        h = h._values
```

Comment:
`h` here is a pandas/cudf. In that case do you want any special case handling here or want cudf to support `._values`?

Reply:
@TomAugspurger I think `h` is always going to be of type int. As hash of any object is expected to be a series of ints, no?
info: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.util.hash_pandas_object.html

Reply:
That's correct. So for that dtype, `.values` will be equivalent to `._values`.

Reply:
Awesome. So, the changes related to this PR are all actually in and hence ready for review. This would help unblock : https://github.com/rapidsai/cudf/pull/2396